### PR TITLE
Verify-webhook-signature

### PR DESCRIPTION
**Pull request by Icegreeen**
## Pull Request Overview

**This PR features two main improvements focused on webhook processing.**

### 1. **Payload Validation**

**WebhookController.php**

- **Functionality Added:** Early validation in the `handleWebhook` method to check if the received payload is well-formed and contains all necessary fields.
- **Impact:** Returns an **HTTP 400** status for malformed or invalid payloads, improving robustness against incorrect data processing.
- **Test Implementation:** A new helper function `invalidWebhookRequest()` was created. This function generates a `Request` object with a JSON payload that deliberately fails to meet expected criteria, aiding in robust error handling testing.

Webhook Payload Validation Test Screenshot

---

### 2. **Header Transformation Functionality**

**VerifyWebhookSignature.php**

- **Enhancement:** Introduced a function `getTransformedHeaders` to simplify the transformation of request headers into an associative array.
- **Benefit:** Streamlines header handling, enhancing clarity and maintainability.

#### Test Implementation
- **Test Setup:** Developed to confirm that the `VerifyWebhookSignature` middleware accurately rejects requests with invalid signatures.
  - **Request Creation:** Simulated a POST request to the `/webhook` endpoint with test data and an intentionally invalid signature header.
  - **Expected Outcome:** Proper rejection of unauthorized access through an `AccessDeniedHttpException`, confirming middleware effectiveness.

Middleware Test Screenshot

@jayanratna

### CODE REVIEW

**Comment by jayanratna** [comment]
Hey @Icegreeen, thank you for creating this ❤️

Apologies I totally missed this. I'll review it sometime this week and get back to you soon.

**Comment by jayanratna** [comment]
Looks good overall. Quick question... Any reason why you updated the way headers are transformed? As both approaches do the same thing.

File: tests/Http/Middleware/VerifyWebhookSignature.php [hunk 1]
```diff
@@ -46,3 +46,16 @@ function withSignature(Request $request, string $secret, int $timestamp): void
         return new Response('OK');
     });
 })->throws(AccessDeniedHttpException::class, 'Message timestamp too new');
+
+test('middleware rejects request with invalid signature', function () {
```
Comment (jayanratna) [inline tests/Http/Middleware/VerifyWebhookSignature.php:50]:
```suggestion
test('it rejects a request with invalid signature', function () {
```

**Comment by Icegreeen** [comment]
> Looks good overall. Quick question... Any reason why you updated the way headers are transformed? As both approaches do the same thing.

By extracting the transformation logic into a separate method (getTransformedHeaders), it is easier to understand and modify the code if needed in the future.

This approach also adheres to the Single Responsibility Principle (S.O.L.I.D.) by separating the header transformation logic from the main request handling logic.

Although this adds more lines of code, it is more organized. But feel free to modify or deprecate it if necessary.

**Review by jayanratna — APPROVED** [review]
Thank you, really appreciate your contribution 🖤

**Pull request merged**